packages/nocodb/src/db/BaseModelSqlv2.ts

```diff
@@ -3582,19 +3582,19 @@ class BaseModelSqlv2 {
           );
         }
       } else {
-        const returningArr: string[] = [];
+        const returningObj: Record<string, any> = {};
 
         if (!raw) {
           for (const col of this.model.primaryKeys) {
-            returningArr.push(col.column_name);
+            returningObj[col.title] = col.column_name;
```

Comment:
Is it possible for us to use Record<string, string> and add a check on loop if required, bc providing undefined or null can break the query

Reply:
oh yes, will do that

Reply:
Actually primary key column_name and title will never be undefinec/null